### ISSUE TITLE
adding more than one custom yolo model in local cvat

### ISSUE THREAD

**Issue by ZK108** [kurkurzz/custom-yolov8-auto-annotation-cvat-blueprint#6]
hello,
Is it possible to add different customized yolov8 models to our local cvat for auto-annotation? (for example one yolov8 model is trained on some specific classes and the other yolov8 model is trained on some other classes?)

thanks

**Comment by kurkurzz** [comment]
Yes it is very much possible. You simply need to have 2 folder of this repo (one using model A, and another one using model B) and run the last command referring to the correct folder for both. 

Just make sure in 'function.yaml' you specify different name between the 2 folder so you would have 2 distinct model running or else it would override each other

**Comment by ZK108** [comment]
Thank you so much for your response.
I tried to follow the steps you provided; with the fact that I’m installing on WSL2.
Actually , I followed the steps, except step 6 which I didn’t know how to build the docker image and also didn’t know what to replace with the path in the command below:
./serverless/deploy_cpu.sh path/to/this/folder/

Instead of this step, after importing the function.yaml file in Nuclio, I went to the Configurations tab there, to the Volumes section, and created a new volume there with this detail:
Name: custom-model
Type: Host path
Mount Path: /opt/nuclio/custom_model.pt
Host Path: <directory>/custom_model.pt

Was it a proper alternative for step 6?


Then about the step 7, unfortunately I don’t get good results:
docker ps --filter NAME=custom-model-yolov8:
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
(there is nothing written below the row)


By the way, as you recommended I Put the “.wslconfig” file (the exact file you provided)  in my C:/Users/<username> folder.

After trying to deploy the model in nuclio:
-	It says that it Failed to deploy
-	Gives this error:
Error - exit status 100 /nuclio/pkg/cmdrunner/shellrunner.go:114 Call stack: stdout: Sending build context to Docker daemon 63.34MB Step 1/19 : FROM ubuntu:24.04 ---> 278628f08d49 Step 2/19 : ARG NUCLIO_LABEL ---> Using cache ---> 93db9c02ba97 Step 3/19 : ARG NUCLIO_ARCH ---> Using cache ---> 35b8a6963616 Step 4/19 : ARG NUCLIO_BUILD_LOCAL_HANDLER_DIR ---> Using cache ---> fe54eaf7a1fe Step 5/19 : ENV DEBIAN_FRONTEND=noninteractive ---> Using cache ---> 9f1b9b3cbd39 Step 6/19 : USER root ---> Running in 7150a7d6c655 ---> Removed intermediate container 7150a7d6c655 ---> 1118cfff0214 Step 7/19 : RUN apt-get update && apt-get -y install curl git python3 python3-pip libgl1-mesa-glx libglib2.0-dev ---> Running in 0093ec03ef89 Get:1 http://security.ubuntu.com/ubuntu noble-security InRelease [126 kB] Get:2 http://archive.ubuntu.com/ubuntu noble InRelease [256 kB] Get:3 http://security.ubuntu.com/ubuntu noble-security/restricted amd64 Packages [598 kB] Get:4 http://archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB] Get:5 http://archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB] Get:6 http://security.ubuntu.com/ubuntu noble-security/multiverse amd64 Packages [15.3 kB] Get:7 http://security.ubuntu.com/ubuntu noble-security/main amd64 Packages [618 kB] Get:8 http://archive.ubuntu.com/ubuntu noble/multiverse amd64 Packages [331 kB] Get:9 http://archive.ubuntu.com/ubuntu noble/restricted amd64 Packages [117 kB] Get:10 http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages [19.3 MB] Get:11 http://security.ubuntu.com/ubuntu noble-security/universe amd64 Packages [725 kB] Get:12 http://archive.ubuntu.com/ubuntu noble/main amd64 Packages [1808 kB] Get:13 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [850 kB] Get:14 http://archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [933 kB] Get:15 http://archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Packages [602 kB] Get:16 http://archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 Packages [18.4 kB] Get:17 http://archive.ubuntu.com/ubuntu noble-backports/universe amd64 Packages [11.9 kB] Fetched 26.6 MB in 1min 40s (266 kB/s) Reading package lists... Reading package lists... Building dependency tree... Reading state information... Package libgl1-mesa-glx is not available, but is referred to by another package. This may mean that the package is missing, has been obsoleted, or is only available from another source [91mE: Package 'libgl1-mesa-glx' has no installation candidate [0m ---> Removed intermediate container 0093ec03ef89 stderr: DEPRECATED: The legacy builder is deprecated and will be removed in a future release. Install the buildx component to build images with BuildKit: https://docs.docker.com/go/buildx/ The command '/bin/sh -c apt-get update && apt-get -y install curl git python3 python3-pip libgl1-mesa-glx libglib2.0-dev' returned a non-zero code: 100 /nuclio/pkg/cmdrunner/shellrunner.go:114 Failed to build /nuclio/pkg/dockerclient/shell.go:119 Failed to build docker image .../pkg/containerimagebuilderpusher/docker.go:70 Failed to build processor image /nuclio/pkg/processor/build/builder.go:267


Despite of this error, the model’s name appears in cvat, and while trying to use it for auto annotation it shows this error: 
Detection error occurred
Failed to parse: http://host.docker.internal:None 

Would you please give me some suggestions about how to overcome these errors. I would greatly appreciate that. My major isn’t computer and as it is obvious I am new to these topics.
Thanks.

p.s.: I modified the function.yaml file in terms of 1- the classes' names, 2- the "version" of ubuntu, ultarlytics, python, numpy and opencv (compatible to the versions i used for training my model).
